alphadia/workflow/peptidecentric/extraction_handler.py

```diff
@@ -333,75 +462,226 @@ def _score_candidates(
 
         return features_df, fragments_df
 
+    def quantify_candidates(
+        self,
+        candidates_df: pd.DataFrame,
+        precursor_fdr_df: pd.DataFrame | None,
+        dia_data: DiaData,  # noqa: F821
+        spectral_library: SpecLibFlat,
+        top_k_fragments: int | None = None,
+    ) -> tuple[pd.DataFrame | None, pd.DataFrame]:
+        """Quantify candidates using classic backend.
+
+        Note: because quantification and scoring are intertwined in the classic backend,
+        this method performs both scoring and quantification, but only returns the fragments dataframe.
 
-class NgExtractionHandler(ClassicExtractionHandler):
+        """
+        del precursor_fdr_df
+
+        features_df_, fragments_df = self.score_and_quantify_candidates(
```

Comment:
I think we also want to return the features here. Otherwise we lose the mapping betwen precursor and fragments. 

The mental model is 
**scoring:**
```
candidates => precursor (with NN features)
```
**quantification:**
```
candidates => precursor (w/o NN features) + quantified fragments
```

The destroyed precursor fragment index trough `flat_frag_start/stop_idx` is most likely the reason the transfer library is malformewd.